M2/Macaulay2/packages/ForeignFunctions.m2

```diff
@@ -516,14 +516,31 @@ SharedLibrary = new SelfInitializingType of BasicList
 SharedLibrary.synonym = "shared library"
 net SharedLibrary := lib -> lib#1
 
+-- on apple silicon machines, shared libraries are often in /opt/homebrew/lib,
+-- but this is not in DYLD_LIBRARY_PATH, so we try there if the first call to
+-- dlopen fails
+if (version#"operating system" == "Darwin" and
+    version#"architecture" == "aarch64")
+then (
+    brewPrefix := replace("\\s+$", "", get "!brew --prefix");
+    dlopen' = filename -> (
+	if first filename == "/" then dlopen filename
```

Comment:
You could potentially use one of these here:
https://github.com/Macaulay2/M2/blob/master/M2/Macaulay2/m2/startup.m2.in#L152-L154

Reply:
Oh nice -- thanks for pointing these out!